pkg/discovery/agent.go

```diff
@@ -18,38 +15,44 @@ import (
 
 // Agent links the mulesoft client and the gateway client.
 type Agent struct {
-	anypointClient      anypoint.Client
-	apicClient          apic.Client
-	apiChan             chan *ServiceDetail
-	assetCache          cache.Cache
-	discoveryIgnoreTags []string
-	discoveryPageSize   int
-	discoveryTags       []string
-	pollInterval        time.Duration
-	publishBufferSize   int
-	stage               string
-	stopAgent           chan bool
-	stopDiscovery       chan bool
-	stopPublish         chan bool
+	anypointClient anypoint.Client
+	assetCache     cache.Cache
+	stage          string
+	stopAgent      chan bool
+	stopDiscovery  chan bool
```

Comment:
stopDisovery and stopPublish can be removed from the Agent struct then right?

Reply:
yeah I think so since the new structs have a stop method